Session tracking docs optimization (#1062)
rudderlabs/rudderstack-docs: docs/sources/event-streams/sdks/session-tracking/manual-session-tracking.mdx

---
title: "Manual Session Tracking"
description: Manually tracking user sessions using the RudderStack SDKs.
---

> RudderStack supports manual session tracking that let you define the start and end of a user session.

Reviewer: ```suggestion
RudderStack supports manual session tracking that lets you define the start and end of a user session.
```

## Manual session tracking methods

The RudderStack SDKs (<Link to="/sources/event-streams/sdks/rudderstack-javascript-sdk/">JavaScript</Link>, <Link to="/sources/event-streams/sdks/rudderstack-android-sdk/">Android</Link>, and <Link to="/sources/event-streams/sdks/rudderstack-ios-sdk/">iOS</Link>) support the following manual session tracking methods:

<table>
  <tr>
    <th width="180px">Method</th>
    <th width="180px">Parameters</th>
    <th>Description</th>
  </tr>
  <tr>
    <td rowspan="2"><code class="inline-code">startSession()</code></td>
    <td>-</td>

>     <td>If no parameter is passed, RudderStack creates a new session and passes the current <code class="inline-code">timestamp</code> as the <Link to="/sources/event-streams/sdks/session-tracking/faq/#how-does-rudderstack-determine-the-sessionid"><code class="inline-code">sessionId</code></Link>.</td>

Reviewer: ```suggestion
    <td>If you don't pass any parameter, RudderStack creates a new session and passes the current <code class="inline-code">timestamp</code> as the <Link to="/sources/event-streams/sdks/session-tracking/faq/#how-does-rudderstack-determine-the-sessionid"><code class="inline-code">sessionId</code></Link>.</td>
```

  </tr>
  <tr>
    <td><code class="inline-code">sessionId</code><br /> <span style="color: #4D4DFF;font-size:12px;">Long integer with minimum length of 10 characters.</span></td>

>     <td> You can pass a custom <code class="inline-code">sessionId</code> and trigger a new user session. <br/><br/> It is <strong>not recommended</strong> to use a decimal number as the <code class="inline-code">sessionId</code>. </td>

Reviewer: ```suggestion
    <td> If you pass the <code class="inline-code">sessionId</code> parameter, RudderStack triggers a new user session. <br/><br/> It is <strong>not recommended</strong> to use a decimal number as the <code class="inline-code">sessionId</code>. </td>
```

  </tr>
  <tr>
    <td><code class="inline-code">endSession()</code></td>
    <td>-</td>
    <td>RudderStack clears the <code class="inline-code">sessionId</code> and ends the session.</td>
  </tr>
</table>

<div class="warningBlock">

> <strong>Manual session tracking overrides <Link to="/sources/event-streams/sdks/session-tracking/#automatic-session-tracking">automatic session tracking</Link></strong>. <br /><br />If automatic session tracking is enabled and you call the <code class="inline-code">startSession()</code> method, then RudderStack disables it <strong>until the app is closed completely</strong>. <br /><br />For more information on the persistence scope of manual session tracking, refer to the <Link to="/sources/event-streams/sdks/session-tracking/faq/#what-is-the-scope-of-persistence-in-case-of-manual-session-tracking">FAQ</Link> guide.

Reviewer: > If automatic session tracking is enabled and you call the <code class="inline-code">startSession()</code> method, then RudderStack disables it <strong>until the app is closed completely

Does it mean that when you call startSession() method, then RudderStack disables the automatic session tracking until the app is closed completely? 

Assuming that manual session tracking overrides the automatic session tracking (which happens by default) whenever a user calls these methods

Author: I used the words `disables it` because `automatic session tracking` is repeated otherwise. Let me think of something to rephrase this in a better way.
